packages/webpack/webpack.config.js

```diff
@@ -190,59 +191,61 @@ console.log('entry', entry)
 
 console.log('webpack plugins', plugins)
 
-// the Kui builder plugin
-plugins.push({
-  apply: compiler => {
-    let hash
-    compiler.hooks.compilation.tap('KuiHtmlBuilder', compilation => {
-      compilation.hooks.afterHash.tap('KuiHtmlBuilder', () => {
-        // we need to inject the name of the main bundle into the configuration
-        hash = compilation.hash
-      })
-    })
-    compiler.hooks.done.tapPromise('KuiHtmlBuilder', () => {
-      // eslint-disable-next-line no-async-promise-executor
-      return new Promise(async (resolve, reject) => {
-        try {
-          const main = contextRoot + `main.${hash}.bundle.js` // <-- this is the name of the main bundle
-          console.log('KuiHtmlBuilder using this build hash', hash)
-
-          const overrides = {
-            build: { writeConfig: false },
-            env: { main, hash, bodyCss: inBrowser ? ['not-electron'] : ['in-electron'] }
-          }
+const clientBase = path.join(stageDir, 'node_modules/@kui-shell/client/config.d')
+const loadOverride = file => {
+  try {
+    return require(path.join(clientBase, file))
+  } catch (err) {
+    // debug('error in loadOverride', err)
+    return {}
+  }
+}
 
-          if (PORT_OFFSET !== undefined) {
-            overrides.env.nameSuffix = PORT_OFFSET
-          }
+const clientOptions = {
+  name: loadOverride('name'),
+  client: loadOverride('client'),
+  style: loadOverride('style'),
+  icons: loadOverride('icons'),
+  opengraph: loadOverride('opengraph')
+}
 
-          if (contentSecurityPolicyForDevServer) {
-            overrides.theme = {
-              // only override the CSP when running webpack-dev-server;
-              // otherwise, we will inherit the settings from theme.json
-              // https://github.com/IBM/kui/pull/2395
-              contentSecurityPolicy: contentSecurityPolicyForDevServer
-            }
-          }
+clientOptions.style.bodyCss = (inBrowser ? ['not-electron'] : ['in-electron']).concat(clientOptions.style.bodyCss)
 
-          if (!inBrowser || isWatching) {
-            overrides.build.buildDir = outputPath
-          }
+if (contentSecurityPolicyForDevServer) {
+  // only override the CSP when running webpack-dev-server;
+  // otherwise, we will inherit the settings from theme.json
+  // https://github.com/IBM/kui/pull/2395
+  clientOptions.client.contentSecurityPolicy = contentSecurityPolicyForDevServer
+}
+
+const htmlBuildOptions = Object.assign(
+  {
+    inject: false
+  },
+  clientOptions,
+  {
+    filename: path.join(outputPath, 'index.html'),
+    template: path.join(stageDir, 'node_modules/@kui-shell/core/templates/index.ejs')
+  }
+)
 
-          // and this will inject it
-          const Builder = require(path.join(builderHome, 'lib/configure'))
-          await new Builder().build(!inBrowser || isWatching ? 'webpack-watch' : 'webpack', overrides)
+console.error('htmlBuildOptions!!!', htmlBuildOptions)
```

Comment:
oops, leftover debugging message

Reply:
also, can't we remove the remaining two files in packages/builder/lib/?

Reply:
webpack/build.sh still refers to index-webpack.html

Reply:
solved, thanks!